app/middlewares/session_restore_error_rescue_middleware.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class SessionRestoreErrorRescueMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    @app.call(env)
+  rescue ActionDispatch::Session::SessionRestoreError
+    req = ActionDispatch::Request.new(env)
```

Comment:
discussed in person: maybe we can try deleting the session instead, if we can get the id

Reply:
Updated code to only delete the session from session store and leave cookies unchanged